app/js/components/service_status.ts

```diff
@@ -0,0 +1,156 @@
+    this.stateColors = {};
+    this.chart = undefined;
+    this.initStates();
+  }
+
+  /**
+   * Init the eventhandlers on the elements
+   */
+  public registerEventHandlers() {
+    this.fetchState();
+  }
+
+  /**
+   * Check if chart is loaded
+   */
+  public isLoaded(): boolean
+  {
+    return this.chart != undefined;
+  }
+
+  private fetchState() {
+    const self = this;
+    $.ajax({
+      url: "/api/service/status/" + this.serviceId,
+      context: document.body
+    }).done(function (e) {
+      self.draw(<StatesResponseJson>e);
+    });
+  }
+
+  private draw(states: StatesResponseJson) {
+    let title = states.service.name;
+    let labels: string[] = [];
+    let tooltips: string[] = [];
+    let data: number[] = [];
+    let backgroundColor: string[] = [];
+
+    let ratio = 100 / Object.keys(states.service.states).length;
+    for (let key in states.service.states) {
+      const state = states.service.states[key];
+
+      data.push(ratio);
+      labels.push(states.service.labels[key]);
+      backgroundColor.push(this.stateColors[state]);
+      tooltips.push(states.service.tooltips[key])
+    }
+
+    let options: Chart.ChartConfiguration = {
+      type: 'doughnut',
+      data: {
+        labels: labels,
+        datasets: [{
+          data: data,
+          backgroundColor: backgroundColor
+        }]
+      },
+      options: {
+        cutoutPercentage: 75,
+        responsive: true,
+        legend: {
+          position: 'right',
+          labels: {
+            fontSize: 16
+          }
+        },
+        title: {
+          fontSize: 20,
+          display: true,
+          text: title,
+          position: 'left',
+        },
+        animation: {
+          animateScale: true,
+          animateRotate: true
+        },
+        tooltips: {
+          callbacks: {
+            label: function(tooltipItem: ChartTooltipItem) {
+              if (tooltipItem.index == undefined) {
+                return '';
+              }
+              return tooltips[tooltipItem.index];
+            }
+          }
+        }
+      }
+    };
+
+    this.chart = new Chart(this.context, options);
+  }
+
+  private initStates() {
+    this.stateColors['danger'] = '#cc0000';
+    this.stateColors['success'] = '#77b300';
+    this.stateColors['info'] = '#ff6600';
+  }
+}
+
+function loadServiceStatus() {
+  if ($('#service-states').length > 0) {
+
+    let elements = document.getElementsByClassName('service-status-canvas');
+    Array.prototype.forEach.call(elements, function (el: HTMLCanvasElement) {
+
+      let serviceId = el.dataset.serviceId;
+      let context = el.getContext("2d");
+
+      if (context == null || serviceId == undefined) {
+        return;
+      }
+
+      const serviceStatus = new ServiceStatus(
+        context,
+        serviceId,
+      );
+      serviceStatus.registerEventHandlers();
+    });
+  }
+}
+
+$(document).ready(loadServiceStatus);
+
+module.exports = loadServiceStatus;
```

Comment:
CRLF

Reply:
I've added the rule to the linter config but it doesn't seem to work. Maybe we could IRL look into this.